lib/Pass/Pass.cpp

```diff
@@ -279,12 +343,14 @@ struct SouperPass : public ModulePass {
       }
       if (!Cand.Mapping.RHS)
         continue;
-      // TODO: add non-const instruction support
+
       Instruction *I = Cand.Origin.getInstruction();
       Instruction *ReplacedInst = Cand.Origin.getInstruction();
+      IRBuilder<> Builder(ReplacedInst->getParent());
+      Builder.SetInsertPoint(ReplacedInst);
```

Comment:
Why not use ```IRBuilder<> Builder(ReplacedInst);```?

Reply:
Also. Why is the builder created here and not in the getValue function?

Reply:
Thanks for this review!

Reply:
Reason for IR builder created outside getValue() function: getValue() is a recursive function, we don't want to create an IR builder in each recursive call.